src/librustdoc/html/markdown.rs

```diff
@@ -1298,35 +1298,42 @@ impl LangString {
                     }
                     LangStringToken::LangToken(x) if extra.is_some() => {
                         let s = x.to_lowercase();
-                        if let Some((flag, help)) = if s == "compile-fail"
-                            || s == "compile_fail"
-                            || s == "compilefail"
-                        {
-                            Some((
+                        if let Some((flag, help)) = match s.as_str() {
+                            "compile-fail" | "compile_fail" | "compilefail" => Some((
                                 "compile_fail",
-                                "the code block will either not be tested if not marked as a rust one \
-                                 or won't fail if it compiles successfully",
```

Comment:
I'm very fine with your suggestion. :)

Reply:
Hum, actually I'm not sure to understand what you have in mind here. For now, help messages are similar but also different.

In your help messages, we lose the information that the code block will likely not be used as test too.

Reply:
So:

> help: this code block may be skipped during testing, because unknown attributes are treated as markers for code samples written in other programming languages, unless it is also explicitly marked as rust

is common to all warnings, right?

Reply:
Ok now it makes sense. Thanks for the clarification!